### PR TITLE
refactor(avails specs)

### PR DESCRIPTION
**Pull request by iPR0D1G1**
fixed: #6490

### CODE REVIEW

File: libs/contract/src/lib/avails/tests/holdbacks.spec.ts [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+  medias: nonCollidingMediaB,
+  duration: collidingDurationA,
+})
+
+/**Holdback[] and term[] */
+const nonCollidingHoldbacks = [
+  nonCollidingHoldback,
+  collidingTerritoriesHoldback,
+  collidingMediasHoldback,
+  // collidingDurationHoldback, will collide with collidingAllTerm due to medias,duration attributes
```
Comment (pldespaigne) [inline libs/contract/src/lib/avails/tests/holdbacks.spec.ts:101]: remove comment

File: libs/contract/src/lib/avails/tests/holdbacks.spec.ts [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+
+/***Medias */
+const nonCollidingMediaA: Media[] = [
+  'payTv', 'freeTv', 'payPerView', 'est', 'nVod', 'aVod', 'fVod',
+]
+const nonCollidingMediaB: Media[] = [
+  'theatrical', 'video', 'planes', 'boats', 'hotels', 'educational', 'rental',
+]
+const collidingMediaA = nonCollidingMediaA
+//collidingMediaB will collide with nonCollidingMediaA
```
Comment (pldespaigne) [inline libs/contract/src/lib/avails/tests/holdbacks.spec.ts:41]: remove comment

File: libs/contract/src/lib/avails/tests/holdbacks.spec.ts [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+const nonCollidingTerritoriesB: Territory[] = [
+  'kyrgyzstan', 'laos', 'latvia', 'lebanon',
+  'lesotho', 'liberia', 'libya', 'liechtenstein',
+  'lithuania', 'luxembourg', 'macao', 'madagascar',
+  'malawi', 'malaysia', 'maldives', 'mali', 'malta',
+  'marshall', 'martinique', 'mauritania', 'mauritius',
+  'mayotte', 'mexico', 'micronesia',
+]
+const collidingTerritoriesA = nonCollidingTerritoriesA
+//collidingTerritoriesB will collide with nonCollidingTerritoriesA
```
Comment (pldespaigne) [inline libs/contract/src/lib/avails/tests/holdbacks.spec.ts:22]: remove comment

File: libs/contract/src/lib/avails/avails.ts [hunk 4]
```diff
@@ -317,32 +317,32 @@ export function getDurationMarkers(mandates: Mandate[], mandateTerms: Term<Date>
 //         HOLDBACKS         //
 // ----------------------------
 
-function collidingDurations(durationA: Duration, durationB: Duration,) {
+
+
+export function collidingDurations(durationA: Duration, durationB: Duration,) {
```
Comment (pldespaigne) [inline libs/contract/src/lib/avails/avails.ts:322]: remove trailing `,`

**Review by pldespaigne — CHANGES_REQUESTED** [review]
`collidingDurations`, `collidingMedias` and `collidingTerritories` are generic function and maybe they could be used outside of holdbacks, maybe for the "IN TERM" functions ?
Could you try that ?

File: libs/contract/src/lib/avails/avails.ts [hunk 5]
```diff
@@ -25,212 +28,86 @@ export interface DurationMarker {
...
-  if (term.medias.length < avail.medias.length) return false;
-  if (avail.medias.some(medium => !term.medias.includes(medium))) return false;
-
-  if (!avail.territories) return false;
-  if (term.territories.length < avail.territories.length) return false;
-  if (avail.territories.some(territory => !term.territories.includes(territory))) return false;
-
-  return true;
+  return !isSameCalendarTerm(term, avail)
+    && avail.medias && allOf(avail.medias).in(term.medias)
```
Comment (GrandSchtroumpf) [inline libs/contract/src/lib/avails/avails.ts:109]: why not using the `optional` here ?

**Review by GrandSchtroumpf — APPROVED** [review]

**Pull request merged**